### ISSUE TITLE
New model MimicMotion_1-1 very heavy!

### ISSUE THREAD

**Issue by SlimeVRX** [Tencent/MimicMotion#27]
I am inferring a video consisting of 72 frames with default parameters, but the processing time has increased significantly!

It takes 3h17p for 72 frames

![image](https://github.com/Tencent/MimicMotion/assets/99313947/0fdcf3ee-93df-49a8-b17f-cb5a89057964)

**Comment by ak01user** [comment]
yes,using a single 4090 gpu is also like this,I replace the 72 to 32,even worse than old ckpt.maybe I did something wrong.

**Comment by zyayoung** [comment]
You can verify if the model is utilizing the GPU effectively. The expected inference speed on an A100 GPU is less than 5 seconds per iteration. However, your reported speed of 251 seconds per iteration appears unusually slow for a 3090 GPU.

**Comment by ak01user** [comment]
@gujiaxi sorry,I just tested a 15 sec long video,It will spend 3h30m.It is too long.

**Comment by SlimeVRX** [comment]
I tried the old model (MimicMotion_1) again and identified that the cause is not the model but the number of frames, 

which is 72 frames, 

Done Pre-process data!
  2%|████▌                                                                                                                                                                                                                               | 1/50 [03:47<3:05:29, 227.14s/it]  
2%|████▌                                                                                                                                                                                                                               | 1/50 [04:26<3:37:50, 266.75s/it]

**Comment by ThatDocBoi** [comment]
Are you sure you're not spilling into system RAM? What resolution & number of frames are you doing?

**Comment by SlimeVRX** [comment]
Yes, 11.2/15.9 share GPU memory this may be the cause of slowness

ckpt_path: models/MimicMotion_1-1.pth
num_frames: 72
resolution: 576

Updated the last commit!
![image](https://github.com/Tencent/MimicMotion/assets/99313947/840ee3bb-3ede-43f7-a60f-ab9555c876d5)

**Comment by ThatDocBoi** [comment]
To prevent this from happening, go to nvidia control panel, under "manage 3d settings" and under "CUDA - System fallback policy" change that setting to "prefer no sysmem fallback" so now you'll just get out of memory errors vs waiting 5 hours.

**Comment by zyayoung** [comment]
This issue is indeed attributed to the system's RAM or out-of-memory (OOM) errors. I am actively working on minimizing the VRAM requirements. I anticipate that the 72-frame model will perform efficiently on 4090.

**Comment by Minamiyama** [comment]
after sync with the latest PR #32  of yesterday night, I also got a long time waiting and large vram size on 4090, ran using `python inference.py --inference_config configs/test.yaml`
![image](https://github.com/Tencent/MimicMotion/assets/918617/78876f5c-75f0-46c1-abcf-095f0cd007ba)
![image](https://github.com/Tencent/MimicMotion/assets/918617/1b3ed8d6-bea7-444f-a7d0-003ace168983)

**Comment by zyayoung** [comment]
> after sync with the latest PR #32 of yesterday night, I also got a long time waiting and large vram size on 4090

@Minamiyama This seems strange. One difference is that I tested on ubuntu, but I don't have a Windows machine with a 4090 GPU for testing. Could you try the following setting?
> To prevent this from happening, go to nvidia control panel, under "manage 3d settings" and under "CUDA - System fallback policy" change that setting to "prefer no sysmem fallback" so now you'll just get out of memory errors vs waiting 5 hours.

Or does anyone else using windows/ubuntu can or cannot run the 72 frame model with 16G VRAM?

**Comment by Minamiyama** [comment]
> > after sync with the latest PR #32 of yesterday night, I also got a long time waiting and large vram size on 4090
> 
> @Minamiyama This seems strange. One difference is that I tested on ubuntu, but I don't have a Windows machine with a 4090 GPU for testing. Could you try the following setting?
> 
> > To prevent this from happening, go to nvidia control panel, under "manage 3d settings" and under "CUDA - System fallback policy" change that setting to "prefer no sysmem fallback" so now you'll just get out of memory errors vs waiting 5 hours.
> 
> Or does anyone else using windows/ubuntu can or cannot run the 72 frame model with 16G VRAM?

yes, as your mention, I put it to docker then, it finally ran totally fine, just using 17 minutes 😄，thx very much

**Comment by akk-123** [comment]
@zyayoung 1.1的效果比1要更差

**Comment by zyayoung** [comment]
@akk-123 You need to set num_frames to 72 for the 1.1 model. The only difference between 1 and 1.1 is the number of frames per segemnt used during training. If you determine that version 1 performs better for your needs, you have the option to use it instead.
